bpf/lib/nodeport.h

```diff
@@ -653,6 +692,16 @@ int tail_nodeport_nat_ipv6(struct __ctx_buff *ctx)
 		goto drop_err;
 	}
 
+	ret = maybe_add_l2_hdr(ctx, DIRECT_ROUTING_DEV_IFINDEX,
+			       &l2_hdr_required);
+	if (ret != 0)
+		goto drop_err;
+	if (!l2_hdr_required)
+		goto out_send;
```

Comment:
nit: the __maybe_unused from `out_send: __maybe_unused` can then probably be removed.